Enemy death

YeDemirkiran opened the pull request:
- The enemy has a death collider set as a triger, placed on the top of itself. 

- EnemyController checks for trigger collisions, and the collider has the tag 'Player', it calls its Kill method, which then calls its implemented OnDeath method. The OnDeath method destroys the object. We can implement the animation there.

YeDemirkiran committed:
Create Enemy prefab and EnemyController scrip

YeDemirkiran committed:
Add enemy death functionality

The enemy has a death collider set as a triger, placed on the top of itself. EnemyController checks for trigger collisions, and the collider has the tag 'Player', it calls its Kill method, which then calls its implemented OnDeath method. The OnDeath method destroys the object. We can implement the animation there.

YeDemirkiran committed:
Add conditions to PlayerController OnDeath

- Added a checker bool 'fellDown' to PlayerController that is set to true, when the death is caused by touching the bottom boundary.

- Player will die only if: fellDown is true, or the player has no armor and is not Titan

zamcham committed:
Create base enemy class

zamcham committed:
Add enemy1 base

zamcham committed:
Add base method of hurting player and enemy death

zamcham committed:
Update player speed and jump settings

zamcham committed:
Add comment

zamcham committed:
Add hurt player method

zamcham committed:
Seriliaze armor bool for testing

zamcham committed:
Create Lumiflute script, basic walker enemy

zamcham committed:
Test walking script on enemy

zamcham committed:
Improve readability

zamcham commented:
## 📝 Overview:

### Refactored Enemy Logic:
- Created a shared "EnemyBase" class to handle enemy death by jumping on it and hurting the player otherwise.
- Added a capsule collider exclusively for jump-on-death functionality.
- Added a BoxCollider2D exclusively for player damage across all enemies.

### Player Controller Enhancement:
- Introduced a "Hurt()" function in the player controller.
- Checks for player armor or being in Titan mode.
- Transforms the player to Nomad form if in Nomad form without armor.

## 🦾 Action Items:

### Scene "Level 1 - The Crater Hideout":
I highly encourage you to work on this scene from now on.

- **To-Do:** Confirm correct player spawn on checkpoint.
- **To-Do:** Ensure a smooth transition for a better player experience. (If not done yet)

### General Development Guidelines:
- **Note:** If you plan to work on something new, create a new branch based on this one.
- **Reminder:** Submit a pull request that merges into this branch instead of the main branch.

YeDemirkiran committed:
Fix infinite respawn on checkpoints

-Due to how IsTouchingLayer method works, it caused a bug that caused inifinte respawns. Now fixed by disabling the ground checker first, and re-enabling it when fully respawned.

YeDemirkiran committed:
Add SetCheckpoint method to PlayerController

This method does what the script has been doing before. But now it also sets the color of the checkpoint to indicate that the checkpoint has been activated.

YeDemirkiran committed:
Create RollingEnemy prefab

- Convert 'EnemyController' to 'RollingEnemy'

- Create RollingEnemy prefab

- Add rolling animation to the prefab

- Create CameraBoundary layer, set it to only collide with the Player layer and add the layer to the camera (if we don't do this, the camera will push the enemies)

YeDemirkiran committed:
Implement Rolling feature for RollingEnemy

- The enemy's sprite rotates according to it's current speed

- It casts two rays and checks whether it is approaching a corner or not. If it is, then it starts to change direction with a coroutine.

YeDemirkiran committed:
Add Recorder package

zamcham committed:
Merge pull request #32 from zamcham/checkpoint-visual-cue

Add visual cue for checkpoint activation

zamcham committed:
Merge pull request #33 from zamcham/rolling-enemy

Add Rolling Enemy
